backend/internal/userschema/handler/userschemahandler.go

```diff
@@ -0,0 +1,347 @@
+	if idValidationFailed {
+		return
+	}
+
+	sanitizedRequest, requestValidationFailed := validateUpdateUserSchemaRequest(w, r, logger, ash)
+	if requestValidationFailed {
+		return
+	}
+
+	updatedUserSchema, svcErr := ash.userSchemaService.UpdateUserSchema(schemaID, sanitizedRequest)
+	if svcErr != nil {
+		handleError(w, logger, svcErr)
+		return
+	}
+
+	if buildUserSchemaResponse(w, updatedUserSchema, logger, http.StatusOK) {
+		return
+	}
+
+	logger.Debug("Successfully updated user schema",
+		log.String("schemaID", schemaID), log.String("name", updatedUserSchema.Name))
+}
+
+// HandleUserSchemaDeleteRequest handles the user schema delete request.
+func (ash *UserSchemaHandler) HandleUserSchemaDeleteRequest(w http.ResponseWriter, r *http.Request) {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, userSchemaLoggerComponentName))
+
+	schemaID, idValidationFailed := extractAndValidateSchemaID(w, r, logger)
+	if idValidationFailed {
+		return
+	}
+
+	svcErr := ash.userSchemaService.DeleteUserSchema(schemaID)
+	if svcErr != nil {
+		handleError(w, logger, svcErr)
+		return
+	}
+
+	w.WriteHeader(http.StatusNoContent)
+	logger.Debug("Successfully deleted user schema", log.String("schemaID", schemaID))
+}
+
+// parsePaginationParams parses limit and offset from query parameters.
+func parsePaginationParams(query map[string][]string) (int, int, *serviceerror.ServiceError) {
+	var limit, offset int
+	var err error
+
+	if limitStr := query["limit"]; len(limitStr) > 0 && limitStr[0] != "" {
+		sanitizedLimit := sysutils.SanitizeString(limitStr[0])
+		limit, err = strconv.Atoi(sanitizedLimit)
+		if err != nil || limit <= 0 {
+			return 0, 0, &constants.ErrorInvalidLimit
+		}
+	}
+
+	if offsetStr := query["offset"]; len(offsetStr) > 0 && offsetStr[0] != "" {
+		sanitizedOffset := sysutils.SanitizeString(offsetStr[0])
+		offset, err = strconv.Atoi(sanitizedOffset)
+		if err != nil || offset < 0 {
+			return 0, 0, &constants.ErrorInvalidOffset
+		}
+	}
+
+	return limit, offset, nil
+}
+
+// handleError handles service errors and converts them to appropriate HTTP responses.
+func handleError(w http.ResponseWriter, logger *log.Logger, svcErr *serviceerror.ServiceError) {
+	var statusCode int
+	if svcErr.Type == serviceerror.ClientErrorType {
+		statusCode = http.StatusBadRequest
+		if svcErr.Code == constants.ErrorUserSchemaNotFound.Code {
+			statusCode = http.StatusNotFound
+		} else if svcErr.Code == constants.ErrorUserSchemaNameConflict.Code {
+			statusCode = http.StatusConflict
+		}
+	} else {
+		statusCode = http.StatusInternalServerError
+	}
+
+	w.Header().Set(serverconst.ContentTypeHeaderName, serverconst.ContentTypeJSON)
+	w.WriteHeader(statusCode)
+
+	errResp := apierror.ErrorResponse{
+		Code:        svcErr.Code,
+		Message:     svcErr.Error,
+		Description: svcErr.ErrorDescription,
+	}
+
+	logger.Error("Request failed", log.String("errorCode", errResp.Code), log.String("errorMessage", errResp.Message))
```

Comment:
we shouldn't log client errors

Reply:
Fixed with https://github.com/asgardeo/thunder/compare/91a4bf9364d05bb4ad158a5fa255b34b3cb1681c..2359bb281523048120acb1f8c4d2b4ce08b10f5f